src/main/java/com/rackspace/salus/telemetry/model/ValidLabelKeysValidator.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Rackspace US, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspace.salus.telemetry.model;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Validates that the annotated map contains that keys that start with a letter and then contain
+ * only letters, digits, and underscore.
+ */
+public class ValidLabelKeysValidator implements ConstraintValidator<ValidLabelKeys, Map<String,String>> {
+   private static final Pattern VALID_KEY = Pattern.compile("[A-Za-z][A-Za-z0-9_]*");
```

Comment:
```suggestion
   private static final Pattern VALID_KEY = Pattern.compile("[A-Za-z0-9_]+");
```

Am I being dumb, or should this not be what's used?

Reply:
Some test cases.

`1_dfw`
`_this_is_a_test`

```
```
 empty string (although, probably fails elsewhere in our code)

Reply:
Your suggested change goes against the javadoc I stated twice:
> keys that start with a letter and then contain only letters, digits, and underscore

but with that said, I don't feel strongly prohibiting the label keys starting with numbers and underscores. I was just following suit of typical programming identifier naming rules.

Reply:
I'll add those as negative test cases...unless you want the leading digit/underscore rule.

Reply:
> I stated twice:

Oh.  My mistake.  I definitely must have skimmed over that.

But I was basing the comment on the validation message that's returned. `All label names must consistent of alpha-numeric or underscore`.  Maybe that needs to be adapted to be clearer.

Reply:
I guess I was thinking there's also no reason to disallow them from starting with those characters?

Reply:
Ah yes, I was avoiding a wordy error message. So I like the idea of keeping the pattern simple.